langchain/agents/agent_toolkits/spark/base.py

```diff
@@ -36,6 +36,82 @@ def create_spark_dataframe_agent(
     if not isinstance(df, DataFrame):
         raise ValueError(f"Expected Spark Data Frame object, got {type(df)}")
 
+    return _create_spark_df_agent_without_validation(
+        llm,
+        df,
+        callback_manager,
+        prefix,
+        suffix,
+        input_variables,
+        verbose,
+        return_intermediate_steps,
+        max_iterations,
+        max_execution_time,
+        early_stopping_method,
+        agent_executor_kwargs,
+        **(kwargs or {}),
+    )
+
+
+def create_spark_connect_dataframe_agent(
```

Comment:
So the only difference between this function and `create_spark_dataframe_agent` is the validation for the input dataframe type. I think we can merge these two functions. Users should use the same method to create a Spark agent, regardless of whether spark connect is enabled.